src/crypto.ts

```diff
@@ -22,14 +22,12 @@ const CIPHER_ALGORITHM = 'aes-256-gcm';
 const ENCRYPTION_RESULT_ENCODING = 'base64';
 
 export interface CryptoOptions {
-  encryptionKey: string;
+  encryptionKey: string | Buffer;
 }
 
-type EncryptInput = string | object | number | boolean;
-
 export interface Crypto {
-  encrypt(input: EncryptInput, aad?: string): Promise<string>;
-  decrypt(output: string | Buffer, aad?: string): Promise<EncryptInput>;
+  encrypt<Input = any>(input: Input, aad?: string): Promise<string>;
+  decrypt<Output = any>(output: string | Buffer, aad?: string): Promise<Output>;
```

Comment:
Checking the implementation of the `decrypt` method, `Output` can only be anything that extends a JSON or the final [`JSON.parse`](https://github.com/elastic/node-crypto/blob/b8b43b9e3884f544c9f4df3aa5d66779b905436a/src/crypto.ts#L150) will fail.

Maybe we want to limit the `Input` to enforce we'll never encrypt anything that is not an object or, even better, we can add a safeguard in the actual implementation to parse the value accordingly (casting via JSON.parse, Number, Boolean or String).

Reply:
Nevermind! The tests prove we do support all the types, so it's essentially `any`:
https://github.com/elastic/node-crypto/blob/b8b43b9e3884f544c9f4df3aa5d66779b905436a/src/crypto.test.ts#L28-L78